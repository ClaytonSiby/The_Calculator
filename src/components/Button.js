import PropTypes from 'prop-types';

const Button = ({ buttonName, clickHandler }) => {
  const handleClick = () => clickHandler(buttonName);
  return (
    <button type="button" className="btnInput" onClick={handleClick}>
      {buttonName}
    </button>
  );
};

Button.propTypes = {
  buttonName: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Button;
