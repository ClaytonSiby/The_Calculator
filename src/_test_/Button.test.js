import { render } from '@testing-library/react';
import Button from '../components/Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
  test('render Button component without errors', () => {
    render(<Button />);
  });

  test('makes sure the output renders consistently', () => {
    const test = renderer.create(<Button />).toJSON();
    expect(test).toMatchSnapshot();
  })
});
