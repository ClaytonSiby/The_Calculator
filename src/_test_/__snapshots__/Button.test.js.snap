// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button makes sure the output renders consistently 1`] = `
<button
  className="maxWidth"
  onClick={[Function]}
  style={
    Object {
      "background": false,
      "width": "25%",
    }
  }
  type="button"
>
  
</button>
`;
